python/shotgun_fields/shotgun_field_delegate.py

```diff
@@ -201,26 +231,28 @@ def setModelData(self, editor, model, index):
         :param editor: The editor widget.
         :type editor: :class:`~PySide.QtGui.QWidget`
         :param model: The SG model where the data lives.
-        :type model: :class:`~shotgun_utils.shotgun_model.ShotgunModel`
+        :type model: :class:`~PySide.QtCore.QAbstractItemModel`
         :param index: The index of the model to be edited.
         :type index: :class:`~PySide.QtCore.QModelIndex`
         """
         src_index = _map_to_source(index)
         if not src_index or not src_index.isValid():
-            # invalide index, do nothing
+            # invalid index, do nothing
             return
 
         new_value = editor.get_value()
 
-        cur_value = src_index.data(shotgun_model.ShotgunModel.SG_ASSOCIATED_FIELD_ROLE)
+        cur_value = src_index.data(self.data_role)
         if cur_value == new_value:
             # value didn't change. nothing to do here.
             return
 
         bundle = sgtk.platform.current_bundle()
 
-        # special case for image fields
-        if editor._field_name == "image":
+        # special case for image fields in the ShotgunModel. The SG model stores
+        # the image field in the first column. If the value has changed, set the
+        # icon value there.
+        if self._source_is_sg_model and editor._field_name == "image":
```

Comment:
Same comment here: is it possible to do images with something else than a ShotgunModel? And should this test moved into some method or property, like `is_image`? The exact same test is used somewhere else.